netsim/modules/lag.yml

```diff
@@ -12,19 +12,21 @@ attributes:
   global:
     lacp: { type: str, valid_values: [ "off", "slow", "fast" ] }
     lacp_mode: { type: str, valid_values: [ "passive", "active" ] }
-    mode: { type: str, valid_values: [ "802.3ad", "balance-xor", "active-backup" ] }
+
+    # All Linux bonding modes that require lag configuration on the switch side
```

Comment:
The only bonding mode that requires switch configuration is "802.3ad" or am I missing something? Why exactly do you think we need Linux bonding modes in the generic LAG module?

Reply:
Based on https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/networking_guide/overview-of-bonding-modes-and-the-required-settings-on-the-switch#overview-of-bonding-modes-and-the-required-settings-on-the-switch

```
Table 7.1. Switch Configuration Settings Depending on the Bonding Modes
Bonding Mode	Configuration on the Switch
0 - balance-rr	Requires static Etherchannel enabled (not LACP-negotiated)
1 - active-backup	Requires autonomous ports
2 - balance-xor	Requires static Etherchannel enabled (not LACP-negotiated)
3 - broadcast	Requires static Etherchannel enabled (not LACP-negotiated)
4 - 802.3ad	Requires LACP-negotiated Etherchannel enabled
5 - balance-tlb	Requires autonomous ports
6 - balance-alb	Requires autonomous ports
```

So 0, 2, 3 and 4 require lag configuration on the switch side, with only 4 including LACP.

1, 5 and 6 are handled by the bonding plugin (server-side config only)